File: output/z_ai_glm_5/zero/cpp/Torre.cpp

#include <iostream>
#include <vector>
#include <algorithm>

using namespace std;

int main() {
    // Otimização de entrada/saída padrão para competitiva
    ios_base::sync_with_stdio(false);
    cin.tie(NULL);

    int n;
    if (!(cin >> n)) return 0;

    // Matriz para armazenar o tabuleiro
    vector<vector<int>> board(n, vector<int>(n));
    
    // Vetores para armazenar a soma de cada linha e de cada coluna
    vector<int> rowSums(n, 0);
    vector<int> colSums(n, 0);

    // Leitura dos dados e cálculo das somas parciais
    for (int i = 0; i < n; ++i) {
        for (int j = 0; j < n; ++j) {
            cin >> board[i][j];
            rowSums[i] += board[i][j];
            colSums[j] += board[i][j];
        }
    }

    int maxWeight = -1;

    // Cálculo do peso máximo
    // Peso(i, j) = (SomaLinha[i] - board[i][j]) + (SomaColuna[j] - board[i][j])
    // Peso(i, j) = SomaLinha[i] + SomaColuna[j] - 2 * board[i][j]
    for (int i = 0; i < n; ++i) {
        for (int j = 0; j < n; ++j) {
            int currentWeight = rowSums[i] + colSums[j] - 2 * board[i][j];
            if (currentWeight > maxWeight) {
                maxWeight = currentWeight;
            }
        }
    }

    cout << maxWeight << endl;

    return 0;
}

File: output/z_ai_glm_5/zero/python/Torre.py

import sys

def main():
    # Lê toda a entrada de uma vez para maior eficiência
    input_data = sys.stdin.read().split()
    
    if not input_data:
        return

    iterator = iter(input_data)
    
    try:
        n_str = next(iterator)
        N = int(n_str)
    except StopIteration:
        return

    # Inicializa a matriz e os arrays de soma
    matrix = []
    row_sums = [0] * N
    col_sums = [0] * N
    
    # Lê a matriz e calcula as somas das linhas e colunas
    for i in range(N):
        row = []
        for j in range(N):
            val = int(next(iterator))
            row.append(val)
            row_sums[i] += val
            col_sums[j] += val
        matrix.append(row)

    max_weight = -1
    
    # Calcula o peso máximo
    # Peso = (Soma da Linha) + (Soma da Coluna) - 2 * Valor da Célula
    # Pois a célula (i, j) está incluída na soma da linha e na da coluna,
    # e a torre não ataca a própria posição onde está.
    for i in range(N):
        for j in range(N):
            weight = row_sums[i] + col_sums[j] - 2 * matrix[i][j]
            if weight > max_weight:
                max_weight = weight
                
    print(max_weight)

if __name__ == "__main__":
    main()
